### PR TITLE
BaseIntegrationTest -> HttpIntegrationTest

### PR DESCRIPTION
**Pull request by alyssawilk**
Moving the bulk of HTTP/H2-specific logic into a subclass of BaseIntegrationTest

Also moving client_protocol_ into the new HttpIntegrationTest

### CODE REVIEW

File: test/integration/integration.cc [hunk 1]
```diff
@@ -356,7 +354,11 @@ void BaseIntegrationTest::createTestServer(const std::string& json_path,
   registerTestServerPorts(port_names);
 }
 
-void BaseIntegrationTest::sendRequestAndWaitForResponse(Http::TestHeaderMapImpl& request_headers,
+HttpIntegrationTest::HttpIntegrationTest(Http::CodecClient::Type client_protocol,
```
Comment (htuch) [inline test/integration/integration.cc]: Do you want to split/rename to http_integation_test.cc etc?

File: test/integration/http2_integration_test.h [hunk 2]
```diff
@@ -5,10 +5,10 @@
 #include "gtest/gtest.h"
 
 namespace Envoy {
-class Http2IntegrationTest : public BaseIntegrationTest,
+class Http2IntegrationTest : public HttpIntegrationTest,
```
Comment (htuch) [inline test/integration/http2_integration_test.h]: While we're thinking about refactoring, protocols and programmatic generation of config, what are your thoughts about being able to independently set the downstream/upstream protocol? Today, upstream is essentially a property of how we setup the config file, rather than test driven. It seems that (just like in GFE), we want to be able to treat these both as first class properties in tests.

Comment (alyssawilk) [reply test/integration/http2_integration_test.h]: Yeah, I think that one may have to want until backend config is proto-driven, but I planned to experiment with it next.

File: test/integration/integration.cc (hunk 1 above)
Comment (alyssawilk) [reply test/integration/integration.cc]: Sure.  I figured diffs would be easier this way, but either way it'll be separate commits.  Will do in the next push :-)

Comment (alyssawilk) [reply test/integration/integration.cc]: Split out but I'll admit I was lazy cleaning up includes and libraries, given it's test code.
Feel free to tell me to be non-lazy if you care about such things, otherwise I'll just make sure the OSX build doesn't choke.

Comment (mattklein123) [reply test/integration/integration.cc]: My preference is to try to keep the test code cleanly as we do with the production code, if you don't mind doing those cleanups. If you are too busy I would potentially just add a TODO for the cleanup.

File: test/integration/BUILD [hunk 3]
```diff
@@ -104,12 +104,26 @@ envoy_cc_test(
...
 
 envoy_cc_test_library(
+    name = "http_integration_lib",
+    srcs = [
+        "http_integration.cc",
+    ],
+    hdrs = [
+        "http_integration.h",
+    ],
+    data = ["//test/common/runtime:filesystem_test_data"],
```
Comment (mattklein123) [inline test/integration/BUILD]: Why is this needed? I don't remember history here.
